crates/swc_ecma_minifier/src/compress/optimize/iife.rs

```diff
@@ -501,13 +501,6 @@ where
                         log_abort!("iife: [x] Found spread argument");
                         return;
                     }
-                    match &*arg.expr {
-                        Expr::Fn(..) | Expr::Arrow(..) => {
```

Comment:
I think we have to check for `this` for `Expr::Fn`

Reply:
I don't think so. Could you provide an example?

Reply:
```js
const obj = {
    a: 1
};
function real() {
    console.log(this.a);
}
(function(foo) {
    console.log(foo.call(obj));
})(function() {
    real.call(this);
});

```

Oh nevermind I was wrong.